modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py

```diff
@@ -36,6 +36,19 @@ def __stringify(t):
     def __membersof(t):
         return [attr for attr in dir(t) if not callable(getattr(t, attr)) and not attr.startswith("_")]
 
+    @staticmethod
+    def __get_last_path_info(fnode):
+        """
+        Returns the last pathInfo added to the file node
+        """
+        if not isinstance(fnode, FileNode):
+            logging.error("'{}' is not an instance of FileNode.".format(fnode))
+            sys.exit(3)
+
+        i = len(fnode.paths)
+        test = fnode.paths[i-1]
+        return fnode.paths[i-1]
```

Comment:
Also check if the filenode is valid first. To be valid it needs to have at least one path. If the length of the paths is zero then an error message should be returned or if the pathInfo.Path is nil.